lib/cn/move.c

```diff
@@ -114,6 +115,25 @@ cn_move(
             cn_tree_samp_update_move(w, src_node);
     }
 
+    /* Unmark input kvsets */
+    {
+        struct kvset_list_entry *le;
+
+        list_for_each_entry(le, &tgt_node->tn_kvset_list, le_link) {
+            bool from_src = kvset_get_nodeid(le->le_kvset) == src_nodeid;
+
+            if (from_src) {
+                assert(kvset_get_workid(le->le_kvset) != 0);
+                kvset_set_workid(le->le_kvset, 0);
+                kvset_set_nodeid(le->le_kvset, tgt_node->tn_nodeid);
+            } else {
+                assert(kvset_get_nodeid(le->le_kvset) == tgt_node->tn_nodeid);
```

Comment:
This doesn't reset the workid of the kvsets in the target node, which is needed for node join.

Reply:
Yeah, I've fixing that.